share/functions/__fish_print_packages.fish

```diff
@@ -35,6 +35,18 @@ function __fish_print_packages
         return
     end
 
+    # pkg_info on OpenBSD provides versioning info which we want for
+    # installed packages but, calling it directly can cause delays in
+    # returning information if another pkg_* tool have a lock.
+    # Listing /var/db/pkg is a clean alternative.
+    if begin
```

Comment:
Again: This `begin; end` pair can be removed.

```fish
if type -q -f pkg_add # which is just `command -sq pkg_add`
   and test (uname) = "OpenBSD"
```

will do the right thing.

I still maintain though that the uname check can be removed as well.